bin/tests/common.rs

```diff
@@ -0,0 +1,80 @@
+use assert_cmd::cargo::CommandCargoExt;
+use core::time;
+use std::fs::OpenOptions;
+use std::io::Write;
+use std::path::PathBuf;
+use std::process::{Child, Command, Stdio};
+use std::sync::mpsc;
+use std::sync::mpsc::TryRecvError;
+use std::thread;
+
+pub struct FileContext {
+    pub file_path: PathBuf,
+    pub stop_handle: Box<dyn FnOnce() -> i32>,
+}
+
+pub fn start_append_to_file(dir: &PathBuf, delay_ms: u64) -> FileContext {
```

Comment:
It's a little nicer to use a Path rather than a pathbuf for function parameters if you don't need it to be mutable.

It's trivial to convert a PathBuf to a path, but converting. Path to a PathBuf requires an allocation and copy (like &str to String)

Reply:
Makes sense, I'll change it.